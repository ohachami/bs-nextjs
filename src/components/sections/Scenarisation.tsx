'use client';
import React from 'react';
import ScenarioCard from '../common/ScenarioCard';
import { Button } from '../ui/button';
import { PlusIcon, Split } from 'lucide-react';
import { useScenarios } from '@/services/scenarios.service';

const Scenarisation: React.FC = () => {
  const { isPending, data: scenarios } = useScenarios();

  if (isPending) return <p>Loading...</p>;
  if (!scenarios) return <p>Error...</p>;

  return (
    <div>
      <div className="flex items-center justify-between">
        <p className="font-semibold text-lg p-4 pl-0"> Liste des scénarios</p>
        <div className="flex gap-4 p-4">
          <Button variant="outline">
            <Split size={16} />
            Comparer
          </Button>
          <Button variant="default">
            <PlusIcon size={16} />
            Ajouter un scénario
          </Button>
        </div>
      </div>
      <div className="grid grid-cols-4 gap-4">
        {scenarios.map((item, index) => (
          <ScenarioCard key={index} scenario={item} />
        ))}
      </div>
    </div>
  );
};

export default Scenarisation;
