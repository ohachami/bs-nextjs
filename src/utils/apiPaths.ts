// List of key;value api paths
export const apiPaths = {
  exerciceTypes: () => `/config/exerciseTypes`,
  periodConfig: () => `/config/periodConfig`,
  periods: () => `/config/periods`,
  stepConfig: () => `/config/steps`,
  exercises: () => `/exercises`,
  exercisesCount: () => `/exercises/count/all`,
  exercisesCountByStatus: (status: string) => `/exercises/count?status=${status}`,
  currentUser: () => `/users/current`,
  users: () => `/users`,
  datasources: (sbuId: string) => `/datasources/hierarchy/${sbuId}`,
  dashboardSections: (stepId: string) => `/dashboard/steps/${stepId}/sections`,
  aggregations: () => `/aggregations`,
  chartList: (sectionId: string) => `/dashboard/sections/${sectionId}`,
  datasourceVersions: (datasourceId: string) =>
    `/datasources/${datasourceId}/versions`,
  version: (id: string) => `/datasources/versions/${id}`,
  regions: () => `/referential/regions`,
  products: () => `/referential/products`,
  productTypes: () => `/referential/product-types`,
  groupedProducts: () => `/referential/grouped-products`,
  sbus: () => `/referential/sbus`,
  consolidationVersions: (sbuId?: string) =>
    !sbuId ? `/consolidation` : `/consolidation?sbuId=${sbuId}`,
  consolidationHierarchy: (sbuId: string, exerciseId?: string) =>
    exerciseId
      ? `/consolidation/hierarchy?sbuId=${sbuId}&exerciseId=${exerciseId}`
      : `/consolidation/hierarchy?sbuId=${sbuId}`,
  scenarios: () => `/scenarios`,
  cloneScenario: (scenarioId: string) => `/scenarios/${scenarioId}/clone`,
  demandTypes: () => `/referential/demand-types`,
  ScenarioParams: (consoDataId: string) => `/scenarios/data?consoDataId=${consoDataId}`,
  consolidationSales: () => `/consolidation/sales`
} as const;
