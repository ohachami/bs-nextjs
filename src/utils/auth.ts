import { getUserPermissions } from '@/db/users.db';
import { AuthOptions, getServerSession } from 'next-auth';
import { generateToken } from './jwt';

const authOptions: AuthOptions = {
  providers: [
    {
      id: 'adfs',
      name: 'ADFS',
      type: 'oauth',
      issuer: process.env.ADFS_ISSUER,
      clientId: process.env.ADFS_CLIENT_ID,
      clientSecret: process.env.ADFS_CLIENT_SECRET,
      authorization: {
        url: `${process.env.ADFS_ISSUER}/oauth2/authorize/`,
        params: {
          scope: 'openid profile email',
        },
      },
      token: {
        url: `${process.env.ADFS_ISSUER}/adfs/oauth2/token/`,
      },
      userinfo: {
        url: `${process.env.ADFS_ISSUER}/userinfo`,
      },
      jwks_endpoint: `${process.env.ADFS_ISSUER}/discovery/keys`, // Optionally for verifying tokens
      checks: ['pkce', 'state'], // Recommended for enhanced security
      profile(profile) {
        return {
          id: profile.sub,
          name: profile.unique_name,
          email: profile.email,
          image: null,
        };
      },
    },
  ],
  session: {
    strategy: 'jwt',
  },
  callbacks: {
    async signIn({ user }) {
      // AD account check
      if (!user.email) {
        throw new Error('Authentication Failed: No Email Found');
      }
      return true;
    },
    async jwt({ token, user, account }) {
      // Persist email across token refreshes
      if (account && user.email) {
        token.adfsToken = account.access_token;
        try {
          // Fetch permissions by email
          const permissions = await getUserPermissions(user.email);
          // Generate custom token with claims
          const customToken = generateToken({ email: user.email, permissions });
          // Update session with token and permissions
          token.accessToken = customToken;
        } catch (error) {
          console.error('Token Generation Error:', error);
          throw new Error('Failed to generate token');
        }
      }
      return token;
    },
    async session({ session, token }) {
      session.accessToken = token.accessToken as string;
      session.adfsToken = token.adfsToken as string;
      return session;
    },
  },

  // Additional options (optional)
  secret: process.env.NEXTAUTH_SECRET, // Ensure you set this environment variable
};

// Extended type definitions
declare module 'next-auth' {
  interface Session {
    user: {
      id: string;
      name: string;
      email: string;
      image: null;
      permissions: string[];
    };
    accessToken: string;
    adfsToken: string;
  }
}

/**
 * Helper function to get the session on the server without having to import the authOptions object every single time
 * @returns The session object or null
 */
const getSession = () => getServerSession(authOptions);

export { authOptions, getSession };
