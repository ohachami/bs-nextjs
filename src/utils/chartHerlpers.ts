import { GroupedData, GroupedDataByRegion, PreSerie } from '@/types/dashboard';

export const makeBoxPlotData = (
  item: Record<string, number>,
  label: string
) => {
  const { MIN = 0, AVG = 0, MAX = 0 } = item;
  return {
    x: label,
    y: [
      Math.ceil(MIN),
      Math.ceil(AVG),
      Math.ceil(AVG),
      Math.ceil(AVG),
      Math.ceil(MAX),
    ],
  };
};

export const transformData = (data: GroupedData): GroupedDataByRegion[] => {
  const groupedData: Record<string, GroupedDataByRegion> = {};

  data.forEach((item) => {
    const label = item.groupedBy.label;
    if (!groupedData[label]) {
      groupedData[label] = {
        label,
        versions: [],
      };
    }
    groupedData[label].versions.push({
      dataVersionId: item.dataVersionId,
      data: item.groupedBy.data,
    });
  });

  return Object.values(groupedData);
};

export const createPreSeriesFromTransformedData = (
  group: GroupedDataByRegion
): PreSerie => {
  const productSet = new Set<string>();
  group.versions.forEach((ver) => {
    ver.data.forEach((product) => {
      productSet.add(product.label);
    });
  });

  const labelsArray = Array.from(productSet);
  const labelsMap: Record<string, number> = {};
  labelsArray.forEach((product, index) => {
    labelsMap[product] = index;
  });

  return {
    name: group.label,
    labels: labelsArray,
    versions: group.versions.map((ver) => {
      const valuesArray: Array<Record<string, number>> = Array(
        labelsArray.length
      ).fill({});

      ver.data.forEach((p) => {
        valuesArray[labelsMap[p.label]] = p.values || { AVG: 0 };
      });

      return {
        dataVersionId: ver.dataVersionId,
        values: valuesArray,
      };
    }),
  };
};
