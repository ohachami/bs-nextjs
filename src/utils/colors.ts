export const ProcessStepColors = {
  INACTIVE_BG: 'transparent',
  INACTIVE_BORDER: '#E2E8F0',
  IN_PROGRESS_BG: '#007bff21',
  IN_PROGRESS_BORDER: '#007BFF',
  DONE_BG: '#cdf1d03b',
  DONE_BORDER: '#57D762',
} as const;
