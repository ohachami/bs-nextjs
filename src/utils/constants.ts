export const modules = {
    marketSteering: 'marketSteering',
    tacticalPlanning: 'tacticalPlanning',
    midTermSupplyChain: 'midTermSupplyChain'
} as const;

// Constant list of permissions excpected for a user
export const permissions = {
    // Administarator Role
    ROLE_ADMIN: "ROLE_ADMIN",
    // User Create Role
    ROLE_USERS_C: "ROLE_USERS_C",
    // User Read Role
    ROLE_USERS_R: "ROLE_USERS_R",
    // User Update Role
    ROLE_USERS_U: "ROLE_USERS_U",
    // User Delete Role
    ROLE_USERS_D: "ROLE_USERS_D",
    ROLE_EXERCISE_LIST_R: "ROLE_EXERCISES_LIST_R",
} as const;

// Exercice Types
export const exerciseTypes = {
    Budget: 'Budget',
    MBR: 'MBR',
    QBR: 'QBR',
    'Ad hoc': 'Ad hoc'
} as const;

// Exercice Status
export const EXERCISE_STATUS = {
    IN_PROGRESS: 'IN_PROGRESS',
    CLOSED: 'CLOSED' 
} as const;

export const STEP_STATUS = {
    DONE: 'DONE',
    INACTIVE: 'INACTIVE',
    IN_PROGRESS: 'IN_PROGRESS',
    CLOSED: 'CLOSED' 
} as const;