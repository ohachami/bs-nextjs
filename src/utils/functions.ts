import { Nullable } from "./types";
import { format } from 'date-fns';
import { fr } from 'date-fns/locale';

export const getFullName = (lastname: Nullable<string>, firstname: Nullable<string>) => {
    let first = '';
    let last = '';

    if (firstname) {
        const trimed = firstname.trim()
        first = trimed.charAt(0).toUpperCase() + trimed.slice(1).toLowerCase();
    }

    if (lastname) {
        last = lastname.trim().toUpperCase();
    }

    return `${first} ${last}`;
}

export const formatDate = (date: Date) => {
    // Format the date
    const formattedDate = format(date, 'dd MMM yyyy', { locale: fr });

    // Capitalize the first letter of the month
    return formattedDate.replace(/(\b[a-z])/i, (char) => char.toUpperCase());

}

export function truncateString(input: Nullable<string>, maxLength: number = 16): Nullable<string> {
    if (!input) {
        return '';
    }
    if (input.length <= maxLength) {
        return input; // Return the string as is if it's shorter than or equal to the max length
    }
    return input.slice(0, maxLength) + ".."; // Truncate and append ".."
}