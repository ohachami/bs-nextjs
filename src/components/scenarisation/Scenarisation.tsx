'use client';

import ScenarioList from './ScenarioList';
import ScenarioParams from './ScenarioParams';
import { useScenarisationStore } from '@/store/exercises/scenarisation';
import { List } from 'lucide-react';
import { Button } from '../ui/button';
import { ChartNoAxesCombined } from 'lucide-react';
import { FileSpreadsheet } from 'lucide-react';
import { Cog } from 'lucide-react';
import { ChevronDown } from 'lucide-react';
import { ChevronUp } from 'lucide-react';
import ModelSettings from './model-settings';
import { ArrowRight } from 'lucide-react';
import { useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';
import {
  FormValues,
  senarioShema,
} from '@/validations/schemas/scenario.schemas';
import { SENARIO_INPUTS_TYPE, SENARIO_TYPE } from '@/utils/constants';
import { Form } from '../ui/form';

const Scenarisation: React.FC = () => {
  const { listScenarios, setlistScenarios, settingModel, setSettingModel } =
    useScenarisationStore();

  const form = useForm<FormValues>({
    resolver: zodResolver(senarioShema),
    mode: 'onChange',
    defaultValues: {
      type: SENARIO_TYPE.EXPLOITATION,
      name: '',
      description: '',
      input: SENARIO_INPUTS_TYPE.DONNEES_CONSOLIDEES,
      consolidated_data_id: '',
      v_min: 0,
      v_max: 0,
      scm_price: 0,
    },
  });

  const id = form.getValues().consolidated_data_id;

  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
  };

  return (
    <Form {...form}>
      <form onSubmit={onSubmit}>
        <div>
          {listScenarios ? (
            <ScenarioList form={form} />
          ) : (
            <div className="flex flex-col gap-y-4">
              <div>
                <Button
                  onClick={setlistScenarios}
                  variant="ghost"
                  className="p-0"
                >
                  <List className="size-4 text-muted-foreground" />
                  Liste des sénarios
                </Button>
              </div>
              <p className="text-2xl font-bold">Scénario Default v1</p>
              <div className="flex items-center justify-between">
                <div className="flex items-center gap-x-5">
                  <div className="text-sm font-normal">Data input</div>
                  <div className="flex gap-x-1 text-sm font-normal">
                    <FileSpreadsheet className="size-4" />
                    Données_consolidées_V2
                  </div>
                  <Button
                    variant={'outline'}
                    className="px-2 py-4"
                    onClick={() => {}}
                  >
                    <ChartNoAxesCombined className="size-4" />
                  </Button>
                </div>

                <Button
                  variant="outline"
                  className="h-8"
                  onClick={setSettingModel}
                >
                  <Cog />
                  Model Setting
                  {settingModel ? <ChevronUp /> : <ChevronDown />}
                </Button>
              </div>

              {settingModel && <ModelSettings />}

              <ScenarioParams consolidated_data_id={id} form={form} />
              <div className="flex justify-end">
                <Button type="submit">
                  Run scénario
                  <ArrowRight className="size-4" />
                </Button>
              </div>
            </div>
          )}
        </div>
      </form>
    </Form>
  );
};

export default Scenarisation;
