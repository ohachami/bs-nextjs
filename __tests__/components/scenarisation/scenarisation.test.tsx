import Scenarisation from '@/components/scenarisation/Scenarisation';
import {
  render,
  waitFor,
  screen,
  fireEvent,
  act,
  RenderResult,
  cleanup,
} from '@testing-library/react';
import { Wrapper } from '../../utils/test-utils';
import userEvent from '@testing-library/user-event';

jest.mock('@/store/exercises/scenarisation', () => ({
  useScenarisationStore: () => ({
    listScenarios: false,
    settingModel: false,
    setlistScenarios: jest.fn(),
    setSettingModel: jest.fn(),
  }),
}));

const mockSubmit = jest.fn((values) => {
  return Promise.resolve({ ...values });
});

describe('Scenarisation component', () => {
  let user: ReturnType<typeof userEvent.setup>;
  let container: RenderResult;

  beforeEach(() => {
    user = userEvent.setup();
    container = render(
      <Wrapper>
        <Scenarisation />
      </Wrapper>
    );
  });

  afterEach(() => {
    cleanup();
  });

  test('renders the componnet and show the correct data', async () => {
    expect(screen.getByText('Scénario Default v1')).toBeInTheDocument();
    expect(screen.getByText('Données_consolidées_V2')).toBeInTheDocument();
    expect(screen.getByText('Data input')).toBeInTheDocument();

    const listScenariosButton = screen.getByTestId('mock-list');
    expect(listScenariosButton).toBeInTheDocument();

    fireEvent.click(listScenariosButton);
    await waitFor(() => {}); // add test when implementation of buttons

    await act(async () => {
      await user.click(
        await screen.findByRole('button', { name: /Model Setting/i })
      );
    });

    await act(async () => {
      await user.click(
        await screen.findByRole('button', { name: /Liste des sénarios/i })
      );
    });

    expect(container).toMatchSnapshot();
  });

  test('should display correct content when clicking tabs', async () => {
    await act(async () => {
      await user.click(await screen.findByText('Inputs'));
    });

    const expectedTexts = [
      'Hypothèses commerciales',
      'Hypothèses Manufacturing',
      'Hypothèses minière',
    ];

    for (const text of expectedTexts) {
      expect(await screen.findByText(text)).toBeInTheDocument();
    }
  });

  test('submit data when clicking run scenario', async () => {
    await act(async () => {
      await user.click(
        await screen.findByRole('button', { name: /Run scénario/i })
      );
    });

    // Wait for the submission process to complete
    await waitFor(() => {
      expect(mockSubmit);
    });
  });
});
