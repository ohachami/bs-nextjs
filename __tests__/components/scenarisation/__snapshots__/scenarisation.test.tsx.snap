// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scenarisation component renders the componnet and show the correct data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div>
          <div
            class="flex flex-col gap-y-4"
          >
            <div>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground h-9 p-0"
              >
                <div
                  data-testid="mock-list"
                />
                Liste des sénarios
              </button>
            </div>
            <p
              class="text-2xl font-bold"
            >
              Scénario Default v1
            </p>
            <div
              class="flex items-center justify-between"
            >
              <div
                class="flex items-center gap-x-5"
              >
                <div
                  class="text-sm font-normal"
                >
                  Data input
                </div>
                <div
                  class="flex gap-x-1 text-sm font-normal"
                >
                  <div
                    data-testid="mock-file-spreadsheet"
                  />
                  Données_consolidées_V2
                </div>
                <button
                  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-9 px-2 py-4"
                >
                  <div
                    data-testid="mock-chart-no-axes"
                  />
                </button>
              </div>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground px-4 py-2 h-8"
              >
                <div
                  data-testid="mock-cog"
                />
                Model Setting
                <div
                  data-testid="mock-chevron-down"
                />
              </button>
            </div>
            <div>
              <div
                data-orientation="horizontal"
                dir="ltr"
              >
                <div
                  aria-orientation="horizontal"
                  class="bg-background gap-2 p-0 pt-2 pb-1 px-1 font-bold rounded-md shadow-none w-full flex justify-start gap-x-5"
                  data-orientation="horizontal"
                  role="tablist"
                  style="outline: none;"
                  tabindex="0"
                >
                  <button
                    aria-controls="radix-:r0:-content-1"
                    aria-selected="true"
                    class="inline-flex items-center justify-center bg-background border-b-2 border-background focus:border-primary ring-0 outline-none shadow-none data-[state=active]:border-b-2 data-[state=active]:border-primary disabled:opacity-100 data-[state=active]:shadow-none rounded-none m-0 pt-1.5 pb-2 hover:bg-background-muted text-sm font-normal w-fit text-muted-foreground px-4"
                    data-orientation="horizontal"
                    data-radix-collection-item=""
                    data-state="active"
                    id="radix-:r0:-trigger-1"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    Paramètres
                  </button>
                  <button
                    aria-controls="radix-:r0:-content-2"
                    aria-selected="false"
                    class="inline-flex items-center justify-center bg-background border-b-2 border-background focus:border-primary ring-0 outline-none shadow-none data-[state=active]:border-b-2 data-[state=active]:border-primary disabled:opacity-100 data-[state=active]:shadow-none rounded-none m-0 pt-1.5 pb-2 hover:bg-background-muted text-sm font-normal w-fit text-muted-foreground px-4"
                    data-orientation="horizontal"
                    data-radix-collection-item=""
                    data-state="inactive"
                    id="radix-:r0:-trigger-2"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    Inputs
                  </button>
                </div>
                <div
                  aria-labelledby="radix-:r0:-trigger-1"
                  class="mt-5 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
                  data-orientation="horizontal"
                  data-state="active"
                  id="radix-:r0:-content-1"
                  role="tabpanel"
                  style="animation-duration: 0s;"
                  tabindex="0"
                >
                  Paramètres compt 
                </div>
                <div
                  aria-labelledby="radix-:r0:-trigger-2"
                  class="mt-5 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
                  data-orientation="horizontal"
                  data-state="inactive"
                  hidden=""
                  id="radix-:r0:-content-2"
                  role="tabpanel"
                  tabindex="0"
                />
              </div>
            </div>
            <div
              class="flex justify-end"
            >
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
                type="submit"
              >
                Run scénario
                <div
                  data-testid="mock-arrow-right"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div>
        <div
          class="flex flex-col gap-y-4"
        >
          <div>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground h-9 p-0"
            >
              <div
                data-testid="mock-list"
              />
              Liste des sénarios
            </button>
          </div>
          <p
            class="text-2xl font-bold"
          >
            Scénario Default v1
          </p>
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex items-center gap-x-5"
            >
              <div
                class="text-sm font-normal"
              >
                Data input
              </div>
              <div
                class="flex gap-x-1 text-sm font-normal"
              >
                <div
                  data-testid="mock-file-spreadsheet"
                />
                Données_consolidées_V2
              </div>
              <button
                class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-9 px-2 py-4"
              >
                <div
                  data-testid="mock-chart-no-axes"
                />
              </button>
            </div>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground px-4 py-2 h-8"
            >
              <div
                data-testid="mock-cog"
              />
              Model Setting
              <div
                data-testid="mock-chevron-down"
              />
            </button>
          </div>
          <div>
            <div
              data-orientation="horizontal"
              dir="ltr"
            >
              <div
                aria-orientation="horizontal"
                class="bg-background gap-2 p-0 pt-2 pb-1 px-1 font-bold rounded-md shadow-none w-full flex justify-start gap-x-5"
                data-orientation="horizontal"
                role="tablist"
                style="outline: none;"
                tabindex="0"
              >
                <button
                  aria-controls="radix-:r0:-content-1"
                  aria-selected="true"
                  class="inline-flex items-center justify-center bg-background border-b-2 border-background focus:border-primary ring-0 outline-none shadow-none data-[state=active]:border-b-2 data-[state=active]:border-primary disabled:opacity-100 data-[state=active]:shadow-none rounded-none m-0 pt-1.5 pb-2 hover:bg-background-muted text-sm font-normal w-fit text-muted-foreground px-4"
                  data-orientation="horizontal"
                  data-radix-collection-item=""
                  data-state="active"
                  id="radix-:r0:-trigger-1"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  Paramètres
                </button>
                <button
                  aria-controls="radix-:r0:-content-2"
                  aria-selected="false"
                  class="inline-flex items-center justify-center bg-background border-b-2 border-background focus:border-primary ring-0 outline-none shadow-none data-[state=active]:border-b-2 data-[state=active]:border-primary disabled:opacity-100 data-[state=active]:shadow-none rounded-none m-0 pt-1.5 pb-2 hover:bg-background-muted text-sm font-normal w-fit text-muted-foreground px-4"
                  data-orientation="horizontal"
                  data-radix-collection-item=""
                  data-state="inactive"
                  id="radix-:r0:-trigger-2"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  Inputs
                </button>
              </div>
              <div
                aria-labelledby="radix-:r0:-trigger-1"
                class="mt-5 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
                data-orientation="horizontal"
                data-state="active"
                id="radix-:r0:-content-1"
                role="tabpanel"
                style="animation-duration: 0s;"
                tabindex="0"
              >
                Paramètres compt 
              </div>
              <div
                aria-labelledby="radix-:r0:-trigger-2"
                class="mt-5 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2"
                data-orientation="horizontal"
                data-state="inactive"
                hidden=""
                id="radix-:r0:-content-2"
                role="tabpanel"
                tabindex="0"
              />
            </div>
          </div>
          <div
            class="flex justify-end"
          >
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
              type="submit"
            >
              Run scénario
              <div
                data-testid="mock-arrow-right"
              />
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
